nav2_bringup/launch/navigation_launch.py

```diff
@@ -59,7 +59,8 @@ def generate_launch_description():
 
     # Create our own temporary YAML files that include substitutions
     param_substitutions = {
-        'autostart': autostart}
+        'autostart': autostart,
+        'topic': (namespace, '/scan')}
```

Comment:
I don't disagree with you necessarily, but this doesn't scale generically to any arbitrary topic for a laser scanner to support multiple laser scanners (or additional sensors like RGBD sensors).

Reply:
> I don't disagree with you necessarily, but this doesn't scale generically to any arbitrary topic for a laser scanner to support multiple laser scanners (or additional sensors like RGBD sensors). This is hardcoded for just one situation - albeit a common one.
> 
> I'm not sure this makes sense to merge for that reason. We need to be careful to be generic so that folks don't run into problems where certain topics or names are "special" and end up with weird states where its hard to tell why some things are working but not others.
> 
> This however might be a sensible thing to add into the multi-robot launch file perhaps? Maybe a commandline argument to set the sensor topics to namespace to make it parameterized. That would then offer the option to use a single config file for the N robots simultaneously that is more generic to different topic name(s) without modifying the normal single-robot case.
> 
> Does that seem reasonable?

Understood what's your point and the confused situation can be occured to the new beginner.

I just tried to merge the changes with minor changes as possible as I can. So I didn't imagine the case of arbitarary topic name for scan topic and multiple laser sensor at that time.
Totally agree that the name should be generic.

Therefore, let me try make another approach. 

My simple idea is just utilizing `ReplaceString()` similiar to 'rviz_luanch.py'
```python
    namespaced_rviz_config_file = ReplaceString(
        source_file=rviz_config_file,
        replacements={'<robot_namespace>': ('/', namespace)})
```

in 'nav2_params.yaml'
```python
local_costmap:
  local_costmap:
    ros__parameters:
      plugins: ["voxel_layer", "inflation_layer"]
      voxel_layer:
        scan:
          topic: <robot_namespace>/scan
```

in 'navigation_launch.py'
```python
    configured_params = RewrittenYaml(
        source_file=params_file,
        root_key=namespace,
        param_rewrites={},
        convert_types=True)
     
    ### if using namespace
    configured_params = ReplaceString(
        source_file=configured_params,
        replacements={'<robot_namespace>': ('/', namespace)})

    ### if not using namespace
    configured_params = ReplaceString(
        source_file=configured_params,
        replacements={'<robot_namespace>': ('')})        
```


What do you think?

Reply:
I think that's another valid way to go, but would require alot of explanation in the documentation that I think may just get lost and result in some confusion (e.g. people just replacing that with their text without thinking much about it) for the single robot case. 

An evolution on that thought is to introduce a second multi-robot launch file which uses a single config file (since right now the multirobot version expects uniquely named config files by the robot's names). In that single file then, we can have the substitution like you suggest since hopefully anyone looking at the multi-robot file will be thinking in a multi-robot fashion to not thoughtlessly delete something that's clearly a substitution point. 

That seems easy enough to get what you're looking for done. The only question is on launch / configuration file naming to distinguish them apart but still be somewhat concise. Maybe add `diff` and `same` in the names? Also open to suggestions

Reply:
OK, creating a separated launch files seems reasonalble.
Then I will modify and make another way base on your opinion.
Just moment please ;)